### PR TITLE
Update technical vision for OpenSSF

### PR DESCRIPTION
**Pull request by bobcallaway**
Per recent discussions at governing board and TAC meetings, the TAC is starting a process to update our technical vision to both:

- help provide greater external clarity on what the foundation wants to achieve in the next few years, and
- help align community & staff against an execution model that helps to deliver on the vision.

This PR aims to start the dialog with some thoughts I've written down based on several conversations over the few weeks; feedback and input from the community is extremely welcome.

Aside: since there can be confusion on the use of the words `{mission,vision,strategy}`, I've included the definitions used below:

- **Mission**: the purpose of the foundation / reason for existence 
    - _Note: the foundation's mission is set by the governing board, not the TAC_
- **Vision**: a description of the desired end state (2-5 years out) 
    - _Note: the TAC owns this deliverable per [foundation charter](https://charter.openssf.org)_
- **Strategy**: logic applied to reach vision

@joshbressers @lukehinds @dlorenc @inferno-chromium @AevaOnline @SecurityCRob FYI

### CODE REVIEW

File: technical-vision.md
Comment (dlorenc) [inline technical-vision.md:19]: The OSSF has played a significant role in public policy discussions this year, which I think has been a success. Should we call those out too?

**Comment by dlorenc** [comment]
Thanks, left one comment!

File: technical-vision.md
Comment (bobcallaway) [reply technical-vision.md:19]: Yes, I see that under the 2nd top-level bullet under advocacy.

Comment (dlorenc) [reply technical-vision.md:19]: Ah ok, I could see it in there too. I was thinking that we'd be a trusted strategic partner for policy folks to reach out to for guidance, but we can also advocate proactively.

File: technical-vision.md
Comment (TheFoxAtWork) [inline technical-vision.md:48]:
```suggestion
### Regardless of (programming language, VCS, CI platform, etc), consumers of OSS can leverage on clear & consistent trusted signals of:
```

Rely implies the work is done by OSS itself, which is not accurate and not representative of the desires of the community, rather that OSS consumers have significant responsibility for consumption of OSS, of which the OpenSSF can assist in defining and integrating those "trusted signals" with projects, **for** consumers.

**Review by TheFoxAtWork — COMMENTED** [review]
What i don't see anywhere within this is shifting the generational gap of cyber security awareness/ skills integration for future potential OSS producers.  The idea of passwords as a security mechanism has risen and fallen over the course of 20+ years - at least one generation. With the current technology space, where Open Source Has Won, there are skills and education that have yet to be defined as well as rise and fall in the next 20 years.

File: technical-vision.md
Comment (torgo) [inline technical-vision.md:63]: Suggest adding something here about raising general awareness within the broader developer community of these issue. This references the “public good” and DEI [values](https://openssf.org/about/values/) - and could underpin activities aimed at under-represented communities.

**Comment by TheFoxAtWork** [comment]
To add - based on chat from TAC call:
guiding behavioral elements within the vision as the method or true north of how the foundation executes on the vision. Terms like "altruistic”, “mutual benefit”, "consensus driven”, "commons" can assist in that
 
commons in this regard per @jchestershopify : as in a shared resource that needs some way to be organised and allocated by consent. Formally commons resources are non-excludable but rivalrous - you can't stop someone from consuming the resource, but if everyone does it purely selfishly the resource collapses (eg fisheries). It's where "tragedy of the commons" comes from

File: technical-vision.md
Comment (bobcallaway) [reply technical-vision.md:48]: The intention behind my use of the word 'rely' was to describe the existence of the signals and their subsequent use in decision making. I did not mean to convey who would be responsible for doing the work to generate those signals. 

I'm not opposed to switching this to "leverage" (as to me it conveys a very similar intended outcome) but I wonder if your point around responsibility is better described in a value statement (that would span the entirety of this vision), similar to @torgo's suggestion below?

Comment (TheFoxAtWork) [reply technical-vision.md:48]: I think a value statement will definitely assist in ensuring the intentions in the vision are relayed accordingly.

Comment (SecurityCRob) [reply technical-vision.md:48]: ### Consumers of OSS can leverage on clear & consistent trusted signals and understand the security profile of open source content:

File: technical-vision.md
Comment (SecurityCRob) [reply technical-vision.md:63]: I suggest something like this perhaps"  ### All consumers of OSS will have access to training, tooling, and security information to make educated choices as they interact with the open source ecosystem."

File: technical-vision.md
Comment (SecurityCRob) [inline technical-vision.md:33]: perhaps add something along these lines  " ### OpenSSF is an influential advocate for efforts aligned with its mission by providing connections to security experts and channels that encourage broad collaboration around its mission"

**Review by SecurityCRob — COMMENTED** [review]
Overall I like this direction and can absolutely get behind these ideas

File: technical-vision.md
Comment (inferno-chromium) [reply technical-vision.md:36]: I couldn't find the advocacy part, but adding "and policy decision makers" seems fine to me.

**Review by inferno-chromium — APPROVED** [review]
Thank you for working on this, looks great.

File: technical-vision.md
Comment (bobcallaway) [reply technical-vision.md:18]: I copied those as-is from the current website, but if folks want to propose changes to those here too I'm supportive of that.

File: technical-vision.md
Comment (TheFoxAtWork) [inline technical-vision.md:14]: How is "improving secure by default" defined in order for advocacy to occur? do we mean to advocate for improving the experience of becoming secure by default? what about changing acceptable security minimums? both?

**Review by TheFoxAtWork — COMMENTED** [review]
Overall LGTM- a few comments to improve clarity and intent.

**Comment by hythloda** [comment]
This is a good step forward I am re-requesting old reviews so that we can get this merged.

File: technical-vision.md
Comment (TheFoxAtWork) [reply technical-vision.md:14]: this still needs addressed. Secure-by-default has a lot of different implementations depending on the context it is used. As written under this header, it simply applies that the entirety of secure-by-default discussions that happen with policy makers will include OpenSSF as a champion for it. None of which speaks the default security behavior, guarantees, or development patterns the OpenSSF's Mobilization Plan, Working Groups, SIGs, etc are working on.

**Review by TheFoxAtWork — CHANGES_REQUESTED** [review]
@hythloda - I reviewed the comments i previously provided and they still stand.

**Review by TheFoxAtWork — CHANGES_REQUESTED** [review]
@hythloda - I reviewed the comments i previously provided and they still stand.

File: technical-vision.md
Comment (bobcallaway) [reply technical-vision.md:10]: Would "users and contributors" be clearer?

File: technical-vision.md
Comment (bobcallaway) [reply technical-vision.md:14]: I'm not sure I understand your concern here. 

I would posit that the OpenSSF is already working on "secure by default" efforts & advocacy (e.g. SLSA that ensures tampering threats to the supply chain are reduced/eliminated; the Memory Safety SIG and entry in the mobilization plan advocates for eliminating entire classes of threats through using memory safe programming languages (e.g. Rust)).

The point of this bullet is to indicate that in N years, if the vision is realized, that  the foundation would have advocated for various actors in the ecosystem to improve their default security posture, and helped to catalyze efforts to make achieving that position easier and with less friction.

Comment (TheFoxAtWork) [reply technical-vision.md:14]: @bobcallaway apologies for the confusion, you addressed it in your response: ([] for additional clarification)
> ensures tampering [and other security] threats to the supply chain, [open source projects, groups, and communities] are reduced/eliminated
> advocate various actors [maintainers, contributors, adopters] in the [open source] ecosystem to improve their default security posture, [development activities], and helped to catalyze efforts to make achieving that position easier and with less friction 

This provides more scoping around secure-by-default specific to the the activities the Foundation chooses to take on. I believe your explanation here, if integrated, would make that intent of that bullet more apparent to someone not already familiar with the OpenSSF.

File: technical-vision.md
Comment (TheFoxAtWork) [reply technical-vision.md:10]: Yes - i think that would be clearer without creating additional work.

**Review by TheFoxAtWork — APPROVED** [review]
LGTM - thank you @bobcallaway

**Comment by SecurityCRob** [comment]
> @SecurityCRob I yield to the new OpenSSF TAC Chair on how to proceed here

I'd like to add this as one of our initial items to get worked out starting next week.

File: technical-vision.md
Comment (lehors) [inline technical-vision.md:9]: I have no problems with that statement but I think it's worth pointing out that this may raise questions at times, OpenVEX being a case in point.

File: technical-vision.md
Comment (lehors) [inline technical-vision.md:16]: I would suggest softening this a bit:
```suggestion
  - Memory Safety efforts that aim to eliminate entire classes of security threats through using memory safe programming languages where possible
```

Comment (SecurityCRob) [reply technical-vision.md:16]: i like this addition, since we can't convert everything.  "where possible" allows us room for alternatives

**Review by SecurityCRob — APPROVED** [review]
I agree with these most recent adjustments to the vision.

File: technical-vision.md
Comment (steiza) [inline technical-vision.md:7]: I may have missed this in an earlier draft, but who are the key stakeholders? OpenSSF membership? The governing board? The open source ecosystem? All of the above?

File: technical-vision.md
Comment (steiza) [reply technical-vision.md:9]: This line initially gave me pause, but I think I've come around to liking it. I'm imagining in practice if the community is not aligned on convergence we'd ask that those projects' readme / documentation include sections on related projects / what's different about their approach.

**Review by steiza — APPROVED** [review]
Minor question, otherwise looks good to me!

File: technical-vision.md
Comment (AevaOnline) [inline technical-vision.md:5]: I suggest replacing the four item list with:

## The OpenSSF is a trusted partner to affiliated open source foundations and projects, and provides valuable services to those projects and foundations.
- The OpenSSF is well positioned within the broader ecosystem and has a clearly articulated charter which supports the activities of the broader community of open source practitioners
- To maintain that trusted relationship, the OpenSSF's mission and value proposition is periodically reviewed with community representatives and constituents, as well as its funding Members
- OpenSSF resources are responsibly managed and utilized towards this Vision in clear, consistent, transparent, and democratic ways.
- Where overlap exists between the OpenSSF's mission and that of another organization, the OpenSSF strives to support other organizations through partnership first, and does not engage in "picking winners and losers".

**Comment by bobcallaway** [comment]
I approve as written as PR submitter (I can't approve my own PR in GH though)

File: technical-vision.md
Comment (hythloda) [reply technical-vision.md:13]: #165

File: technical-vision.md
Comment (bobcallaway) [reply technical-vision.md:30]: I'd always presumed that this was the case, but I agree it would be good to make this explicit.

File: technical-vision.md
Comment (lehors) [reply technical-vision.md:13]: How often should this periodic review take place? This doesn't necessarily have to be answered here but we ought to have that documented somewhere or I'm afraid it won't get implemented.

Comment (bobcallaway) [reply technical-vision.md:13]: The foundation publishes an annual report, but my 2c: the frequency of meetings needs to be quarterly, perhaps with a more frequent newsletter to drive awareness of all the good things going on within the community.

Comment (SecurityCRob) [reply technical-vision.md:13]: I think, depending on which stakeholder, quarterly could be achievable (i.e. - checking in with the GB).  I don't know we have the resources or capabilities to, for example, talk to "oss consumers" with that high a frequency without setting up surveys of townhalls/office hours/etc.  In TAC PR165(https://github.com/ossf/tac/issues/165) I put my initial thoughts on our key stakeholder groups.  My thoughts and conversation welcome, and once we land on the groups, we could fine-tune cadence/channels to engage them.

Comment (SecurityCRob) [reply technical-vision.md:13]: > The foundation publishes an annual report, but my 2c: the frequency of meetings needs to be quarterly, perhaps with a more frequent newsletter to drive awareness of all the good things going on within the community.

I also like the newsletter idea, summarizing activities and upcoming items.

Comment (lehors) [reply technical-vision.md:13]: > The foundation publishes an annual report, but my 2c: the frequency of meetings needs to be quarterly, perhaps with a more frequent newsletter to drive awareness of all the good things going on within the community.

Sorry, which meetings are you referring to?

Comment (bobcallaway) [reply technical-vision.md:13]: The meetings to review the value prop with stakeholders referred to by this bullet, but those meetings are just an idea at this point.

File: technical-vision.md
Comment (di) [inline technical-vision.md:14]: Joint OKRs with who? I'm assuming this is referring to the "large OSS foundations" in the title and not the "individual OSS projects".

**Review by di — COMMENTED** [review]
Overall LGTM, just some small comments.

File: technical-vision.md
Comment (bobcallaway) [reply technical-vision.md:19]: I think it was in reference to "use of actionable standards"... MFA is a de-facto standard across the industry and recognized as better than just passwords, so OpenSSF should support folks using it (as we did in the great MFA token giveaway)

File: technical-vision.md
Comment (bobcallaway) [reply technical-vision.md:14]: yes, it was written under the assumption that the larger OSS foundations might desire and explicitly agree to having shared OKRs. 

If an individual project maintainer for a project deemed as critical wanted to have a shared roadmap with the OpenSSF to drive improvement in their security posture in a way that needed high-touch engagement from the foundation, I would think the foundation would be supportive of that.

**Pull request merged**
